src/redfish/messages/messages.py

```diff
@@ -0,0 +1,182 @@
+    Raised when an operation has failed (HTTP Status >= 400)
+    """
+    pass
+
+class RedfishPasswordChangeRequiredError( Exception ):
+    """
+    Raised when password change required
+    """
+    def __str__(self):
+        return  "\n{}\nURL: {}\n".format( str(self.args[0]), str(self.args[1]) )
+
+def get_messages_detail( response ):
+    """
+    Builds messages detail dict in the payload
+
+    Args:
+        response: The response to parser
+
+    Returns:
+        The dict containing messages_detail
+        messages_detail["status"]: http status code
+        messages_detail["successful"]: response successful (http status code < 400)
+        messages_detail["code"]: redfish message response code field
+        messages_detail["@Message.ExtendedInfo"]: redfish message response code field
+    """
+
+    messages_detail = {}
+    messages_detail["status"] = response.status
+    messages_detail["successful"] = False
+    messages_detail["@Message.ExtendedInfo"] = []
+
+    if response.status >= 400:
+        messages_detail["successful"] = False
+    else:
+        messages_detail["successful"] = True
+
+    try:
+        message_body = response.dict
+        if not "@Message.ExtendedInfo" in message_body:
+            message_body = response.dict["error"]
+        check_message_field = True
+        if "@Message.ExtendedInfo" in message_body:
+            messages_detail["@Message.ExtendedInfo"] = message_body["@Message.ExtendedInfo"]
+            for index in range(len(messages_detail["@Message.ExtendedInfo"])):
+                messages_item = messages_detail["@Message.ExtendedInfo"][index]
+                if not "MessageId" in messages_item:
+                    messages_item["MessageId"] = ""
+                if not "@odata.type" in messages_item:
+                    messages_item["@odata.type"] = ""
+                if not "RelatedProperties" in messages_item:
+                    messages_item["RelatedProperties"] = []
+                if not "Message" in messages_item:
+                    messages_item["Message"] = ""
+                if not "MessageArgs" in messages_item:
+                    messages_item["MessageArgs"] = []
+                if not "Severity" in messages_item:
+                    messages_item["Severity"] = ""
+                if not "MessageSeverity" in messages_item:
+                    messages_item["MessageSeverity"] = ""
+                if not "Resolution" in messages_item:
+                    messages_item["Resolution"] = ""
+                messages_detail["@Message.ExtendedInfo"][index] = messages_item
+                check_message_field = False
+
+        if check_message_field is True:
+            messages_detail["@Message.ExtendedInfo"] = []
+            messages_item = {}
+            if "code" in message_body:
+                messages_item["MessageId"] = message_body["code"]
+            else:
+                messages_item["MessageId"] = ""
+            if "message" in message_body:
```

Comment:
Like with the above for loop, I don't think it's good to insert data that's not provided back by the service.

In the `error` object, `code` is mandatory, but like with MessageId, the lack of it will break clients. So, having a default blank string is safer at the moment.